src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/temperature/NodeLevelDimensionalSummary.java

```diff
@@ -381,4 +351,11 @@ public JsonElement toJson() {
             return summaryObj;
         }
     }
+
+    private class ShardProfileComparator implements Comparator<ShardProfileSummary> {
+        @Override
+        public int compare(ShardProfileSummary o1, ShardProfileSummary o2) {
```

Comment:
nit: A comment would be nice here, this seems "backwards" until you realize we're trying to sort temperatures from high to low

Reply:
Makes sense. Will add one

Reply:
LGTM!